map_model/src/pathfind/mod.rs

```diff
@@ -237,29 +237,32 @@ impl Path {
         self.steps.push_back(step);
     }
 
-    // TODO This is a brittle, tied to exactly what opportunistically_lanechange does.
-    pub fn approaching_uber_turn(&self) -> bool {
-        if self.steps.len() < 5 || self.uber_turns.is_empty() {
-            return false;
-        }
-        if let PathStep::Turn(t) = self.steps[1] {
-            if self.uber_turns[0].path[0] == t {
-                return true;
-            }
-        }
-        if let PathStep::Turn(t) = self.steps[3] {
-            if self.uber_turns[0].path[0] == t {
-                return true;
-            }
-        }
-        false
+    pub fn is_upcoming_uber_turn_component(&self, t: TurnID) -> bool {
+        self.uber_turns
+            .front()
+            .map(|ut| ut.path.contains(&t))
+            .unwrap_or(false)
     }
 
     /// Trusting the caller to do this in valid ways.
     pub fn modify_step(&mut self, idx: usize, step: PathStep, map: &Map) {
         assert!(self.currently_inside_ut.is_none());
         assert!(idx != 0);
         self.total_length -= self.steps[idx].as_traversable().length(map);
+
+        // When replacing a turn, also update any references to it in uber_turns
+        if let PathStep::Turn(old_turn) = self.steps[idx] {
+            for uts in &mut self.uber_turns {
```

Comment:
This is kind of ham-fisted. Since, at the call-site we know something about the offset of our turn it's possible we could do something smarter, but it got a little complicated. I profiled this and it didn't contribute any meaningful runtime, so I decided to keep it like this.